mtg_card_identifier/CardScanner/CardScannerView.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  CardScannerView.swift
+//  mtg_card_identifier
+//
+//  Created by mac on 8/16/23.
+//
+
+import SwiftUI
+import VisionKit
+import AVFoundation
+
+struct CardScannerView: View {
+    @State private var cameraAccessAuthorized = false
+
+    var body: some View {
+        NavigationView {
+            if cameraAccessAuthorized {
+                CardScanner()
+            } else {
+                NoCameraAccessView()
+            }
+        }
+        .onAppear {
+            requestCameraAccess()
+            checkCameraAuthorizationStatus()
+        }
+    }
+
+    func requestCameraAccess() {
+        AVCaptureDevice.requestAccess(for: .video) { granted in
+            if granted {
+                print("Camera access granted")
+                cameraAccessAuthorized = true
+            } else {
+                print("Camera access denied")
+                cameraAccessAuthorized = false
+            }
+        }
+    }
+    
+    func checkCameraAuthorizationStatus() {
+            let authorizationStatus = AVCaptureDevice.authorizationStatus(for: .video)
+            
+            switch authorizationStatus {
+            case .authorized:
+                cameraAccessAuthorized = true
+            case .notDetermined, .denied, .restricted:
+                cameraAccessAuthorized = false
+            @unknown default:
+                cameraAccessAuthorized = false
+            }
+        }
+    }
+
+struct CardScannerView_Previews: PreviewProvider {
+    static var previews: some View {
+        CardScannerView()
+    }
+}
+
+
+struct CardScanner: View {
```

Comment:
I might pull this into the CardScannerView and get rid of CardScanner.

Reply:
So I pulled it into its own file would that be better or should i go ahead and put it in the LiveTextScanner or in the CardScannerView?